vali_objects/scoring/scoring.py

```diff
@@ -33,54 +33,59 @@ def compute_results_checkpoint(
-        )
+        scoring_config = Scoring.get_scoring_config(positions=full_positions, evaluation_time_ms=evaluation_time_ms)
 
-        # Compute miner penalties
         miner_penalties = Scoring.miner_penalties(filtered_positions, ledger_dict)
 
         # Miners with full penalty
         full_penalty_miner_scores: list[tuple[str, float]] = [
             (miner, 0) for miner, penalty in miner_penalties.items() if penalty == 0
         ]
-        full_penalty_miners = set([x[0] for x in full_penalty_miner_scores])
+        # Run all scoring functions and apply penalties
+        penalized_scores_dict = Scoring.score_and_penalize(ledger_dict=ledger_dict,
+                                                           positions=full_positions,
+                                                           scoring_config=scoring_config,
+                                                           evaluation_time_ms=evaluation_time_ms)
 
-        scoring_config = {
-            'return_long': {
-                'function': Scoring.risk_adjusted_return,
-                'weight': ValiConfig.SCORING_RETURN_LONG_LOOKBACK_WEIGHT,
-                'positions': filtered_positions
-            },
-            'return_short': {
-                'function': Scoring.risk_adjusted_return,
-                'weight': ValiConfig.SCORING_RETURN_SHORT_LOOKBACK_WEIGHT,
-                'positions': filtered_recent_positions
-            },
-            'sharpe_ratio': {
-                'function': Scoring.sharpe,
-                'weight': ValiConfig.SCORING_SHARPE_WEIGHT,
-                'positions': filtered_positions
-            },
-            'omega': {
-                'function': Scoring.omega,
-                'weight': ValiConfig.SCORING_OMEGA_WEIGHT,
-                'positions': filtered_positions
-            },
-        }
+        # Combine penalized scores
+        combined_scores  = Scoring.combine_scores(penalized_scores_dict)
 
-        combined_scores = {}
+        # Force good performance of all error metrics
+        combined_weighed = Scoring.softmax_scores(list(combined_scores.items())) + full_penalty_miner_scores
+        combined_scores = dict(combined_weighed)
 
+        # Normalize the scores
+        normalized_scores = Scoring.normalize_scores(combined_scores)
+        return sorted(normalized_scores.items(), key=lambda x: x[1], reverse=True)
+    
+    @staticmethod
+    def score_and_penalize(
+            ledger_dict: dict[str, PerfLedgerData],
+            positions: dict[str, list[Position]],
+            scoring_config: dict[str, dict],
+            evaluation_time_ms: int= None):
+        
+        if evaluation_time_ms is None:
+            evaluation_time_ms = TimeUtil.now_in_millis()
+
+        # Compute miner penalties
+        miner_penalties = Scoring.miner_penalties(positions, ledger_dict)
+
+        # Miners with full penalty
+        full_penalty_miners: list[tuple[str, float]] = set([
```

Comment:
type hint for full_penalty_miners should be set[str], but can also just be omitted